core_v2/sources/v2_domains.move

```diff
@@ -0,0 +1,1318 @@
+    ///
+    /// NOTE: Registration validation already done. This function does not perform any validation on whether `sign` is allowed to register this name
+    fun register_name_internal(
+        sign: &signer,
+        subdomain_name: Option<String>,
+        domain_name: String,
+        registration_duration_secs: u64,
+        price: u64
+    ) acquires CollectionCapability, NameRecord, SubdomainExt, RegisterNameEvents, ReverseRecord, SetReverseLookupEvents {
+        // If we're registering a name that exists but is expired, and the expired name is a primary name,
+        // it should get removed from being a primary name.
+        clear_reverse_lookup_for_name(subdomain_name, domain_name);
+
+        let name_expiration_time_secs = timestamp::now_seconds() + registration_duration_secs;
+
+        // if it is a subdomain, and it expires later than its domain, throw an error
+        // This is done here so that any governance moderation activities must abide by the same invariant
+        if (is_subdomain(subdomain_name)) {
+            let domain_record = get_record(domain_name, option::none());
+            assert!(
+                name_expiration_time_secs <= domain_record.expiration_time_sec,
+                error::out_of_range(ESUBDOMAIN_EXPIRATION_PASS_DOMAIN_EXPIRATION)
+            );
+        };
+
+        // If the token already exists, transfer it to the signer
+        // Else, create a new one and transfer it to the signer
+        let account_addr = signer::address_of(sign);
+        let token_addr = get_token_addr_inline(domain_name, subdomain_name);
+        if (object::is_object(token_addr)) {
+            let record = borrow_global_mut<NameRecord>(token_addr);
+            record.expiration_time_sec = name_expiration_time_secs;
+            record.target_address = option::none();
+            record.registration_time_sec = timestamp::now_seconds();
+            object::transfer_with_ref(object::generate_linear_transfer_ref(&record.transfer_ref), account_addr);
+        } else {
+            create_token(
+                account_addr,
+                domain_name,
+                subdomain_name,
+                name_expiration_time_secs,
+            );
+        };
+
+        event::emit_event<RegisterNameEvent>(
+            &mut borrow_global_mut<RegisterNameEvents>(@aptos_names_v2).register_name_events,
+            RegisterNameEvent {
+                domain_name,
+                subdomain_name,
+                registration_fee_octas: price,
+                expiration_time_secs: name_expiration_time_secs,
+            },
+        );
+    }
+
+    // === RENEW DOMAIN ===
+
+    public fun renew_domain(
+        sign: &signer,
+        domain_name: String,
+        renewal_duration_secs: u64,
+    ) acquires CollectionCapability, NameRecord, SubdomainExt, RenewNameEvents, ReverseRecord {
+        // check the domain eligibility
+        let length = validate_name_string(domain_name);
```

Comment:
this doesn't seem necessary? This should already been checked as part of registering the domain the first time

Reply:
OK